src/Mod/TechDraw/App/DrawViewPart.cpp

```diff
@@ -720,6 +787,35 @@ void DrawViewPart::dumpVertexes(const char* text, const TopoDS_Shape& s)
     }
 }
 
+void DrawViewPart::countFaces(const char* text, const TopoDS_Shape& s)
+{
+    TopExp_Explorer expl(s, TopAbs_FACE);
+    int i;
+    for (i = 0 ; expl.More(); expl.Next(),i++) {
+    }
+    Base::Console().Message("COUNT - %s has %d Faces\n",text,i);
+}
+
+void DrawViewPart::countWires(const char* text, const TopoDS_Shape& s)
+{
+     TopExp_Explorer expl(s, TopAbs_WIRE);
+     int i = 0;
+     for (; expl.More(); expl.Next()) {
+         i++;
+     }
+     Base::Console().Message("COUNT - %s has %d wires\n",text,i);
+}
+
+void DrawViewPart::countEdges(const char* text, const TopoDS_Shape& s)

```

Comment:
Be careful that (unless it's intended) this doesn't count the correct number of edges. The explorer works recursively, i.e. when iterating over a shape element it also iterates over its children and thus an edge will be counted several times.

Example:
A cube is 12 edges but this method counts 24 because the cube has 6 faces with four edges each => 6 * 4 = 24.
If you want to determine the correct number you must do it this way:
`
TopTools_IndexedMapOfShape mapOfEdges;
TopExp::MapShapes(myShape, TopAbs_EDGE, mapOfEdges);
int num = mapOfEdges.Extent();
`
The same applies to countFaces & countWires.